### PR TITLE
Refactor FileRepository

### PR DESCRIPTION
**Pull request by niezbop**
- Extract .unitypackage extraction from FileRepository to a new class UnityPackageOpener.

- Various cleaning of calls inside FileRepository

### CODE REVIEW

File: Assets/Plugins/Editor/Uplift/Common/UnityPackageOpener.cs [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+using SharpCompress.Archives;
+using SharpCompress.Archives.Tar;
+using UnityEngine;
+using Uplift.Extensions;
+
+namespace Uplift.Common
+{
+    public class UnityPackageOpener
+    {
+        public void OpenUnityPackage(string archivePath, bool deleteAfterOpening = false)
```
Comment (lacostej) [inline Assets/Plugins/Editor/Uplift/Common/UnityPackageOpener.cs]: Would `ExtractUnityPackage` be a better name?

File: Assets/Plugins/Editor/Uplift/Common/UnityPackageOpener.cs [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+        {
+            OpenUnityPackage(
+                archivePath,
+                Path.GetDirectoryName(archivePath),
+                deleteAfterOpening
+            );
+        }
+        public void OpenUnityPackage(string archivePath, string destinationPath, bool deleteAfterOpening = false)
```
Comment (lacostej) [inline Assets/Plugins/Editor/Uplift/Common/UnityPackageOpener.cs]: We should say what happens when `destinationPath` doesn't exist.

Comment (lacostej) [inline Assets/Plugins/Editor/Uplift/Common/UnityPackageOpener.cs]: Is `deleteAfterOpening` needed? What's the use case?

Comment (niezbop) [reply Assets/Plugins/Editor/Uplift/Common/UnityPackageOpener.cs]: Documentation-wise?

Comment (niezbop) [reply Assets/Plugins/Editor/Uplift/Common/UnityPackageOpener.cs]: None actually. Removing it

Comment (lacostej) [reply Assets/Plugins/Editor/Uplift/Common/UnityPackageOpener.cs]: Yes

**Pull request merged**